pkg/types/core/oracle_factory.go

```diff
@@ -0,0 +1,21 @@
+package core
+
+import (
+	"context"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+)
+
+type OracleFactory interface {
+	NewOracle(ctx context.Context, args OracleArgs) (offchainreporting2plus.Oracle, error)
```

Comment:
Modify this to return a "custom Oracle" that takes in Start(ctx) and Close(ctx) to preserve ctx.